proposals/0012-f3-deals-reliability-ux.md

```diff
@@ -0,0 +1,159 @@
+5. One-stop set of JSON-RPC operations that encapsulate the new behaviour with no frills (without leaking implementation details as much as possible).
+
+**Technical reliability track**
+
+0. Do a full inventory of outstanding issues; get to the bottom of the long-standing mystical issues; analyse; fix. Some known tasks:
+1. Testing: Design and implement test suites to reproduce the concrete issues and serve as regression tests. Nightly tests for deals.
+2. Fix memory footprint issues.
+3. Fix resumability of data transfers.
+4. Consider p2p file-oriented transfer protocol to cut out complexity for now.
+5. Identify areas of the deals flow code that need more informative logging (eg data transfer restarts / connection retries).
+6. Add logs and journal events.
+7. Fix miner datastore bloat.
+
+#### What does done look like?
+
+_What specific deliverables should completed to consider this project done?_
+
+1. A new version of Lotus (and associated modules) that achieves the success criteria.
+2. A new interactive CLI command (`lotus store`) that guides the user through the process of performing a storage deal, and provides visibility and a polished flow.
+3. A simplified, porcelain S3-like JSON-RPC API that covers the deals flow and provides the logical sequence of steps outlined in previous sections, under a refined, well-documented, and opaque API that doesn't leak Filecoin implementation details.
+4. (Optionally) A prototype for a deals dashboard web UI (in collaboration with The Spark).
+
+#### What does success look like?
+
+_Success means impact. How will we know we did the right thing?_
+
+- 80% of deals initiated via the CLI result in success, as long as prerequisites are met (enough balance, synced node, etc.) under a variety of conditions (e.g. connection losses, slow bandwidth, concurrent deals).
+- Most open issues related to deals on the Filecoin issue tracker are closed, and not reopened, and new ones related to deals are not created shortly thereafter (i.e. we don't introduce new problems).
+- Users give consistent positive feedback about the usability and visibility characteristics of deal-making.
+
+#### Counterpoints & pre-mortem
+
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+N/A. There's pretty good consensus that this is a high impact project.
+
+#### Alternatives
+
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need?_
+
+N/A.
+
+#### Dependencies/prerequisites
+
+Possibly https://github.com/protocol/web3-dev-team/pull/9.
+
+#### Future opportunities
+
+We won't be focusing on the deals dashboard web UI right off the bat, but we could work with The Spark to deliver a prototype, and then open up a bounty/grant for full development.
+
+## Required resources
+
+#### Effort estimate
+
+Large/XLarge, depending on final scope and solution design.
```

Comment:
This is my concern. Can we boil down to just “large” by getting really clear on what’s in scope? Concerned this secretly has 3-4 separate projects buried in it

Reply:
It does, but I think we're better off creating a phased milestone approach to an XLarge project, than breaking it up into interdependent smaller projects just to represent progression. Some things can be broken out:

* Deals web dashboard (I'll submit a pitch for this).
* Deals end-to-end testing (#28).